fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/wrapper/ScannerWrapperV2.kt

```diff
@@ -207,16 +198,22 @@ internal class ScannerWrapperV2(
         }
     }
 
-    override suspend fun turnOnFlashingWhiteSmileLeds(): Unit = withContext(ioDispatcher) {
-        scannerV2.setSmileLedState(scannerUiHelper.whiteFlashingLedState()).onErrorComplete().await()
+    override suspend fun turnOnFlashingWhiteSmileLeds(): Unit = executeSafely {
+        // No need to handle exceptions while updating the LED state
+        runCatching { scannerV2.setSmileLedState(scannerUiHelper.whiteFlashingLedState()) }
     }
 
-    override suspend fun setUiGoodCapture(): Unit = withContext(ioDispatcher) {
-        scannerV2.setSmileLedState(scannerUiHelper.goodScanLedState()).onErrorComplete().await()
+    override suspend fun setUiGoodCapture(): Unit = executeSafely {
+        // No need to handle exceptions while updating the LED state
+        runCatching { scannerV2.setSmileLedState(scannerUiHelper.goodScanLedState()) }
     }
 
+    override suspend fun setUiBadCapture(): Unit = executeSafely {
+        // No need to handle exceptions while updating the LED state
+        runCatching { scannerV2.setSmileLedState(scannerUiHelper.badScanLedState()) }
+    }
 
-    override suspend fun setUiBadCapture(): Unit = withContext(ioDispatcher) {
-        scannerV2.setSmileLedState(scannerUiHelper.badScanLedState()).onErrorComplete().await()
+    companion object {
+        private const val LIVE_FEEDBACK_DELAY_MS = 200L
```

Comment:
Where did this value come from?

Reply:
In the previous implementation, there was no explicit delay between scanner LED updates. The app relied on the natural delays introduced by switching between coroutine dispatchers and Rx schedulers. After removing Rx, the updates became much faster, leading the app to send excessive updates to the scanner. To address this, I added a 200ms delay to ensure that only 5 updates are sent per second. This frequency is subtle enough not to be noticeable by the user while preventing the scanner from being overwhelmed.